### PR TITLE
Support Server documentation

### PR DESCRIPTION
**Pull request by psladek**

### CODE REVIEW

**Comment by lnussel** [comment]
this looks very specific to https://github.com/os-autoinst/os-autoinst-distri-opensuse/tree/master/data/supportserver, maybe should be added there?

**Comment by psladek** [comment]
you mean directly into tests/opensuse/data/supportserver, as a separate .asciidoc file ?

**Comment by lnussel** [comment]
yes, as it's quite specific to that code

**Comment by psladek** [comment]
hm, I thought the original idea was to merge it to openQA doc, where it should extend the section on advanced examples (some more specific usage examples are yet to be added). Also I need to reference some things in previous chapters, that would not be so convenient in completely separated location...

File: docs/WritingTests.asciidoc [hunk 1]
```diff
@@ -416,3 +416,127 @@ sub run {
...
+SUPPORT_SERVER_ROLES=dhcp,tftp
+WORKER_CLASS=server,qemu_autoyast_tap_64
+--------------------------------------------------------------------------------
+
+with a test-suites name +supportserver-opensuse-tftp+.
+
+The actual test 'child' job, will then have to set +PARALLEL_WITH=supportserver-opensuse-tftp+, and also other variables according to the test requirements. For convenience, we have also started a dhcp server on the supportserver, but even without it, network could be set up manually by assigning a free ip address (e.g. 10.0.2.15) on the system of the test job.
+
+The code in the *.pm module doing the actual tftp test could then look something like this:
+
```
Comment (aaannz) [inline docs/WritingTests.asciidoc]: [source,perl]

File: docs/WritingTests.asciidoc [hunk 2]
```diff
@@ -416,3 +416,127 @@ sub run {
...
+--------------------------------------------------------------------------------
+use strict;
+use base 'basetest';
+use testapi;
+
+
+
+
+--------------------------------------------------------------------------------
```
Comment (aaannz) [inline docs/WritingTests.asciidoc]: there really isn't any `sub run {}` ?

File: docs/WritingTests.asciidoc [hunk 3]
```diff
@@ -416,3 +416,127 @@ sub run {
...
+
+
+
+Example - autoyast based tftp test:
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Here we will use autoyast to setup the system of the test job and the os-autoinst autoyast testing infrastructure. For supportserver, this means using proxy to access qemu provided data, for dowloading autoyast profile and tftp verify script:
+
+--------------------------------------------------------------------------------
```
Comment (aaannz) [inline docs/WritingTests.asciidoc]: I think [source,ini] should work here to get some basic highlight

File: docs/WritingTests.asciidoc [hunk 4]
```diff
@@ -416,3 +416,127 @@ sub run {
...
+DESKTOP=textmode
+INSTALLONLY=1
+PARALLEL_WITH=supportserver-opensuse-tftp
+--------------------------------------------------------------------------------
+
+again assuming the support server's name being +supportserver-opensuse-tftp+. Note that the +pxe+ role already contains +tftp+ and +dhcp+ server role, since they are needed for the pxe boot to work.
+
+The tftp test defined in the +autoyast_opensuse/opensuse_autoyast_tftp.sh+ file could be something like:
+
+--------------------------------------------------------------------------------
```
Comment (aaannz) [inline docs/WritingTests.asciidoc]: [source,sh]

File: docs/WritingTests.asciidoc [hunk 5]
```diff
@@ -416,3 +416,127 @@ sub run {
...
+
+where the +DISTRI+, +VERSION+, +FLAVOR+ and +ARCH+ correspond to the job group containing the tests.
+Note that the networking is provided by tap devices, so both jobs should run on machines defined by (apart from others) having +NICTYPE=tap+, +WORKER_CLASS=qemu_autoyast_tap_64+.
+
+
+Example - a simple tftp test:
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Let's assume that we want to test tftp client operation. For this, we setup the supportserver as a tftp server:
+[source,sh]
```
Comment (aaannz) [inline docs/WritingTests.asciidoc]: [source,ini] I think

**Comment by aaannz** [comment]
So what about this? It has been stalling for quite a long time. I agree that it is a highly specific to support server, but who will it read there ( `os-autoinst-distri-opensuse/tests/support_server/` ). So maybe it's not a bad thing to let it live in here.
Maybe add a header that this documents opensuse support server tests (to illustrate what all you can do in openqa) and also to inform reader it's not available as part of testapi etc.

**Comment by coolo** [comment]
I'm find with adding all documentations here - as long it's clear what it's about

**Comment by coolo** [comment]
I wonder why the problems Ondřej highlighted in october aren't fixed yet

**Pull request merged**